.github/workflows/run_gtest.yml

```
@@ -0,0 +1,434 @@
+      build-system: ${{ steps.detect.outputs.build-system }}
+      cpp-standard: ${{ steps.detect.outputs.cpp-standard }}
+      compiler: ${{ steps.detect.outputs.compiler }}
+      source-files: ${{ steps.detect.outputs.source-files }}
+      test-files: ${{ steps.detect.outputs.test-files }}
+      include-dirs: ${{ steps.detect.outputs.include-dirs }}
+      has-custom-main: ${{ steps.detect.outputs.has-custom-main }}
+      project-structure: ${{ steps.detect.outputs.project-structure }}
+    
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
```

actionlint (1.7.7)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)